### DISCUSSION TITLE
Can I pass variables form JavaScript file to tailwind.config file? (I am using tailwind with reactjs)

### DISCUSSION THREAD

**Discussion by b-meet** [tailwindlabs/tailwindcss#9241] [Help]
I am using `tailwind CSS` for first time in my project. I know how to use `tailwind-config` to make variables for `colors`, `border-radius` etc.

but In this scenario,

The `background-color`, `color` etc. will come through `API` so the response is set into a `state` variable.

Now I actually need to add the value of `state` variable into in `tailwind-config` so that the variable for `background-color`, `color` etc in `tailwind-config` gets its value and due to that every element using it changes accordingly.

**Comment by wongjn** [comment]
Tailwind is run at build time but I would presume you would retrieve your colors through the API at runtime and thus, it would not be possible to pass your colors from the API directly to the Tailwind config. Instead, you could set your colors directly on elements with the `style` attribute or if you needed Tailwind's capabilities with regards to variants, you could [set up colors as CSS variables](https://tailwindcss.com/docs/customizing-colors#using-css-variables) and pass through the "real" color values from the API at runtime.

**Reply by b-meet** [reply]
The problem is now resolved thank you for the response it was really helpful. I faced other issue of passing a variable with value of color which came through API in CSS and [this stack overflow answer](https://stackoverflow.com/questions/68812903/how-to-use-api-color-variable-in-css-file/68909117#68909117) helped me out of it
